TTS/tts/models/vits.py

```diff
@@ -1046,18 +1422,32 @@ def get_criterion(self):
             VitsGeneratorLoss,
         )
 
-        return [VitsGeneratorLoss(self.config), VitsDiscriminatorLoss(self.config)]
+        return [VitsDiscriminatorLoss(self.config), VitsGeneratorLoss(self.config)]
 
     def load_checkpoint(
-        self, config, checkpoint_path, eval=False
+        self,
+        config,
+        checkpoint_path,
+        eval=False,
+        strict=True,
     ):  # pylint: disable=unused-argument, redefined-builtin
         """Load the model checkpoint and setup for training or inference"""
         state = torch.load(checkpoint_path, map_location=torch.device("cpu"))
         # compat band-aid for the pre-trained models to not use the encoder baked into the model
         # TODO: consider baking the speaker encoder into the model and call it from there.
         # as it is probably easier for model distribution.
         state["model"] = {k: v for k, v in state["model"].items() if "speaker_encoder" not in k}
-        self.load_state_dict(state["model"])
+        # handle fine-tuning from a checkpoint with additional speakers
+        if hasattr(self, "emb_g") and state["model"]["vits.emb_g.weight"].shape != self.emb_g.weight.shape:
+            num_new_speakers = self.emb_g.weight.shape[0] - state["model"]["vits.emb_g.weight"].shape[0]
+            print(f" > Loading checkpoint with {num_new_speakers} additional speakers.")
+            emb_g = state["model"]["vits.emb_g.weight"]
+            new_row = torch.randn(num_new_speakers, emb_g.shape[1])
+            emb_g = torch.cat([emb_g, new_row], axis=0)
+            state["model"]["vits.emb_g.weight"] = emb_g
```

Comment:
Good idea :).  I will use this approach in languages as well. It plan do that in other PR (when we start to explore the YourTTS language transfer).

Reply:
👍